tests/unit/io_queue_test.cc

```diff
@@ -510,3 +513,54 @@ SEASTAR_TEST_CASE(test_request_iovec_split) {
 
     return make_ready_future<>();
 }
+
+SEASTAR_THREAD_TEST_CASE(test_tb_params) {
+    internal::disk_config_params disk_config(1);
+    internal::disk_params d;
+    size_t iops_req_size = 512;
+    size_t bw_req_size = 128 << 10; // 128k
+
+    // Test multipl datapoints starting at 1GB/s for read/write bandwidth
+    // and 15k req/s for read/write IOPS
+    for (uint64_t i = 1; i < 65; ++i) {
+        d.read_bytes_rate = i << 30;  // iGB/s
+        d.write_bytes_rate = i << 30;
+        d.read_req_rate = i * 15000;
+        d.write_req_rate = i * 15000;
+
+        auto io_config = disk_config.generate_config(d, 0, 1);
+        io_config.mountpoint = std::to_string(i);
+
+        io_queue_for_tests tio(io_config);
+
+        auto cost_read_512 = tio.queue.request_capacity(internal::io_direction_and_length(internal::io_direction_and_length::read_idx, iops_req_size));
+        auto cost_write_512 = tio.queue.request_capacity(internal::io_direction_and_length(internal::io_direction_and_length::write_idx, iops_req_size));
+
+        auto cost_read_128k = tio.queue.request_capacity(internal::io_direction_and_length(internal::io_direction_and_length::read_idx, bw_req_size));
+        auto cost_write_128k = tio.queue.request_capacity(internal::io_direction_and_length(internal::io_direction_and_length::write_idx, bw_req_size));
+        seastar_logger.info("{} read req/s, {} write req/s, {} read bytes/s, {} write bytes/s",
+                            d.read_req_rate, d.write_req_rate, d.read_bytes_rate, d.write_bytes_rate);
+
+        seastar_logger.info("Read 512 cost: {}, Write 512 cost: {}", cost_read_512, cost_write_512);
+        seastar_logger.info("Read 128k cost: {}, Write 128k cost: {}", cost_read_128k, cost_write_128k);
+
+        const auto& fg = internal::get_throttler(tio.queue, internal::io_direction_and_length::write_idx);
+        const auto& tb = fg.token_bucket();
+        double rate = tb.rate();
+        std::cerr << "TB rate: " <<  rate << std::endl;
```

Comment:
Better to log using seastar_logger

Reply:
remainder from debugging, sorry, I'll remove.